tests/api2/test_306_chart_release.py

```diff
@@ -0,0 +1,457 @@
+    assert result.status_code == 200, result.text
+
+
+def test_31_set_custom_catalog_for_testing_update():
+    global results
+    payload = {
+        'force': False,
+        'preferred_trains': ['charts'],
+        'label': 'UPDATECHARTS',
+        'repository': 'https://github.com/ericbsd/charts-1.git',
+        'branch': 'master'
+    }
+    results = POST('/catalog/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    # the sleep is needed or /catalog/items is not ready on time.
+    time.sleep(1)
+
+
+@pytest.mark.parametrize('key', list(updatechart_catalog.keys()))
+def test_32_verify_updatechart_catalog_object(key):
+    assert results.json()[key] == updatechart_catalog[key], results.text
+
+
+def test_33_verify_updatechart_is_in_catalog_list():
+    results = GET('/catalog/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list), results.text
+    assert 'UPDATECHARTS' in results.text, results.text
+
+
+@pytest.mark.parametrize('key', list(updatechart_catalog.keys()))
+def test_34_verify_updatechart_catalog_object(key):
+    results = GET('/catalog/id/UPDATECHARTS/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()[key] == updatechart_catalog[key], results.text
+
+
+def test_35_get_plex_old_version():
+    global old_plex_version, new_plex_version
+    results = POST('/catalog/items/', {'label': 'UPDATECHARTS'})
+    old_plex_version = sorted(list(results.json()['charts']['plex']['versions'].keys()))[0]
+    new_plex_version = sorted(list(results.json()['charts']['plex']['versions'].keys()))[-1]
+
+
+@pytest.mark.dependency(name='release_plex')
+def test_36_create_plex_chart_release_with_old_version(request):
+    depends(request, ['setup_kubernetes'], scope='session')
+    global plex_id
+    payload = {
+        'catalog': 'UPDATECHARTS',
+        'item': 'plex',
+        'release_name': 'plex',
+        'train': 'charts',
+        'version': old_plex_version
+    }
+    results = POST('/chart/release/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), int), results.text
+    job_status = wait_on_job(results.json(), 300)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    plex_id = job_status['results']['result']['id']
+
+
+def test_37_get_plex_chart_release_upgrade_summary(request):
+    depends(request, ['release_plex'])
+    global update_version
+    results = POST('/chart/release/upgrade_summary/', {'release_name': 'plex'})
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert new_plex_version in results.json()['latest_version'], results.text
+    update_version = results.json()['latest_version'].split('_')[1]
```

Comment:
Please remove the split once the related PR has been merged ( https://github.com/truenas/middleware/pull/6694 )

Reply:
Done and tested.